### PR TITLE
Ckan 2.9 upgrade dataset page UI

### PR DESCRIPTION
**Pull request by rleblanc-on**
### What this PR accomplishes
[DATA-546](https://ontariodigital.atlassian.net/browse/DATA-546)
- Moves left-hand items to the right side
- Removes the images associated with ministries
- Removes the “Groups” tab from the navigation bar
- Moves the “Additional Information” section above the resources
- Suppresses empty metadata fields in the "Additional Information" section.
- Removes the green box with checkmark that says "Data available"
- Changes the heading Data to the dataset name
- Adds a ministry contact link at the bottom of the dataset description - (DATA-632)
- **Note that the format of the date ranges have not been changed**

### What still needs to be done
- There are a few files where I have commented out previous code instead of removing it.
- Translations have not been added to the .po / .pot files, nor has it been compiled.

### CODE REVIEW

**Comment by KatiRG** [comment]
Based on figma, there should be a horizontal line between resources of different date ranges, but there is no line:

Screen Shot 2023-03-16 at 14 28 09

**Comment by KatiRG** [comment]
- [ ]      * Moves left-hand items to the right side

- [ ]      * Removes the images associated with ministries

- [ ]      * Removes the “Groups” tab from the navigation bar

- [ ]      * Moves the “Additional Information” section above the resources

- [ ]      * Suppresses empty metadata fields in the "Additional Information" section.

- [ ]     * Removes the green box with checkmark that says "Data available"

- [ ]      * Changes the heading Data to the dataset name
confused about this, because if the resource is missing a name and it is type dataset, the heading "Data" is the default that will be displayed

- [ ]      * Adds a ministry contact link at the bottom of the dataset description - (DATA-632)
=> missing french translation

**Comment by KatiRG** [comment]
The first row "Tags" in the Additional Information table is wider than the rest of the rows:
Screen Shot 2023-03-16 at 14 47 28

File: ckanext/ontario_theme/fanstatic/common.css [hunk 1]
```diff
@@ -1081,6 +1193,143 @@ via submission form. Copied from DS.*/
...
+}
+
+.dataset-font{
+  font-size: 19px;
+  line-height: 25.6px;
+  font-weight: 400;
+}
+.right-col{
+  float: right;
+  max-width: 350px;
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1243]: unset max-width on small screens so the aside is full width

File: ckanext/ontario_theme/templates/internal/package/read.html [hunk 2]
```diff
@@ -1,8 +1,28 @@
-{% ckan_extends %}
+{% extends "package/read_base.html" %}
 
+{% block primary %}
+<article class="module package-z-index">
+  <header class="page-header col-sm-8">
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/templates/internal/package/read.html:5]:
```suggestion
  <header class="page-header col-xs-12 col-sm-8">
```
To allow aside to be full width

File: ckanext/ontario_theme/templates/external/scheming/package/read.html [hunk 3]
```diff
@@ -29,31 +28,33 @@ <h2>{% trans %}For more information{% endtrans %} </h2>
...
-    {% endif %}
-    {% if contact_page %}
+  <div class="ministry-contact">
+	<b>{% trans %}Ministry contact: {% endtrans %} </b>
+	{% if contact_email %}
+	  {% if contact_name %}
+      <a href="mailto:{{contact_email}}?subject={{pkg.title}}">
+        {{contact_name}}
+      </a>
+	  {% endif %}
```
Comment (KatiRG) [inline ckanext/ontario_theme/templates/external/scheming/package/read.html:38]: can combine the two `if` statements into one and just use `else` to distinguish if name or no name.

File: ckanext/ontario_theme/templates/internal/package/read.html [hunk 4]
```diff
@@ -11,6 +31,73 @@
...
+    {% endblock %}
+  
+    {% block package_tags %}
+      {% snippet "package/snippets/tags.html", tags=pkg.tags %}
+    {% endblock %}
+  
+    <div class="ontario-callout ontario-border-highlight--purple">
+      <h2 class="ontario-callout__title ontario-h5">{{ _("Report an error with this data") }}</h2>
+      <p>{% trans %}Fill out a <a href="https://www.ontario.ca/form/report-data-error">Data Catalogue error report</a> to let us know if there are issues with the resources in this dataset.{% endtrans %}</p>
+    </div>
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/templates/internal/package/read.html:91]:
```suggestion
    {% snippet 'package/snippets/report_an_error.html' %}
```

**Review by Alexandra-Green — CHANGES_REQUESTED** [review]
Part 1 of review

**Comment by KatiRG** [comment]
The Additiona Info table headers don't size well on mobile:
Screen Shot 2023-03-16 at 15 37 57

File: ckanext/ontario_theme/templates/internal/package/read.html [hunk 5]
```diff
@@ -1,8 +1,28 @@
-{% ckan_extends %}
+{% extends "package/read_base.html" %}
 
+{% block primary %}
+<article class="module package-z-index">
+  <header class="page-header col-sm-8">
+    <div class="content_action">
+      {% block content_action %}
+      {% if h.check_access('package_update', {'id':pkg.id }) %}
```
Comment (KatiRG) [inline ckanext/ontario_theme/templates/internal/package/read.html:8]: indent

File: ckanext/ontario_theme/templates/internal/package/read.html [hunk 6]
```diff
@@ -1,8 +1,28 @@
...
+      {% link_for _('Edit'), named_route='dataset.edit', id=pkg.name, class_='btn btn-secondary', icon='wrench' %}
+      {% endif %}
+      {% endblock %}
+    </div>
+    <ul class="nav nav-tabs">
+      {% block content_primary_nav %}
+      {{ h.build_nav_icon(dataset_type ~ '.read', _('Dataset'), id=pkg.id if is_activity_archive else pkg.name) }}
```
Comment (KatiRG) [inline ckanext/ontario_theme/templates/internal/package/read.html:15]: indent the block contents

File: ckanext/ontario_theme/templates/internal/package/read.html [hunk 7]
```diff
@@ -11,6 +31,73 @@
...
+      {% endblock %}
+    </h1>
+    {% block package_notes %}
+      {% if pkg.notes %}
+        <div class="notes embedded-content dataset-font">
+          {{ h.render_markdown(h.get_translated(pkg, 'notes')) }}
+        </div>
+      {% endif %}
+    {% endblock %}
+    {# FIXME why is this here? seems wrong #}
```
Comment (KatiRG) [inline ckanext/ontario_theme/templates/internal/package/read.html:71]: ?

File: ckanext/ontario_theme/fanstatic/common.css [hunk 8]
```diff
@@ -1081,6 +1193,143 @@ via submission form. Copied from DS.*/
...
+}
+section.additional-info table.table tbody tr:nth-child(even) th,
+section.additional-info table.table tbody tr:nth-child(even) td {
+  background-color: #f9f9f9;
+}
+section.additional-info table.table tbody tr th:first-child,
+section.additional-info table.table tbody tr td:first-child{
+  padding-top: 10px;
+}
+section.additional-info table.table tbody tr th:last-child,
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1287]: This bottom border is not actually being added.

**Review by Alexandra-Green — CHANGES_REQUESTED** [review]
Part 2 of review

- Clicking on the activity stream tab, the main section of the page is pushed to the bottom of the page because of the aside. You'll need to change `col-sm-9` to `col-sm-8`


Screenshot from 2023-03-16 15-19-40 

- File language wrapping and margin issues. 

Screenshot from 2023-03-16 14-41-29

- resource description and ministry contact font styling do not match Figma

File: ckanext/ontario_theme/fanstatic/common.css (hunk 8 above)
Comment (Alexandra-Green) [reply ckanext/ontario_theme/fanstatic/common.css:1287]: Actually none of the bottom borders in the th column are equal to the td bottom borders

**Comment by rleblanc-on** [comment]
> The first row "Tags" in the Additional Information table is wider than the rest of the rows: Screen Shot 2023-03-16 at 14 47 28

@Alexandra-Green I'm going to need your help on this one. Do you know where this is defined? It doesn't seem to be in `additional_info.html`. Could it be `templates/internal/scheming/display_snippets/fluent_tags.html` , and the classes referenced there?

File: ckanext/ontario_theme/templates/internal/package/read.html (hunk 7 above)
Comment (rleblanc-on) [reply ckanext/ontario_theme/templates/internal/package/read.html]: Holdover from the CKAN Core. I removed it.

File: ckanext/ontario_theme/templates/internal/package/read.html (hunk 2 above)
Comment (rleblanc-on) [reply ckanext/ontario_theme/templates/internal/package/read.html]: Added as part of the solution above, thanks!

File: ckanext/ontario_theme/fanstatic/common.css (hunk 1 above)
Comment (rleblanc-on) [reply ckanext/ontario_theme/fanstatic/common.css]: This seems to have done the trick, though I'm not sure if it's what you would have done:
```
.right-col{
  float: right;
}

@media screen and (min-width: 73em) {
  .right-col {
    max-width: 350px;
  }
}
```

**Comment by rleblanc-on** [comment]
> table headers don't size well on mobile:

@Alexandra-Green I'll need your help with this one tomorrow, too.

File: ckanext/ontario_theme/templates/internal/package/read.html [hunk 9]
```diff
@@ -1,16 +1,92 @@
...
+          {% if h.check_access('package_update', {'id':pkg.id }) %}
+            {% link_for _('Edit'), named_route='dataset.edit', id=pkg.name, class_='btn btn-secondary', icon='wrench' %}
+          {% endif %}
+        {% endblock %}
+      </div>
+      {# Override CKAN: Do not pass any icons to nav bar menu items #}
+      <ul class="nav nav-tabs">
+        {% block content_primary_nav %}
+          {{ h.build_nav_icon(dataset_type ~ '.read', _('Dataset'), id=pkg.id if is_activity_archive else pkg.name) }}
+            <!--{{ h.build_nav_icon(dataset_type ~ '.groups', _('Groups'), id=pkg.id if is_activity_archive else pkg.name) }}-->
```
Comment (KatiRG) [inline ckanext/ontario_theme/templates/internal/package/read.html:17]: Groups should be put back in

**Comment by rleblanc-on** [comment]
Designers have confirmed that RSS heading text should be 'Keep updated', which has been added and translated
image

File: ckanext/ontario_theme/fanstatic/common.css [hunk 10]
```diff
@@ -1081,6 +1193,178 @@ via submission form. Copied from DS.*/
...
+}
+
+.resource-list {
+  overflow: hidden;
+}
+
+.resource-item {
+  padding-top: 15px;
+	padding-bottom: 15px;
+}
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1208]:
```suggestion
.resource-item {
  padding: 15px 30px 15px 30px;
}
.label-div {
  width: 13%;
  float: left;
  position: relative;
  margin: 5px;
}
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 11]
```diff
@@ -1081,6 +1193,178 @@ via submission form. Copied from DS.*/
...
+  font-weight: 400;
+}
+.right-col{
+  float: right;
+}
+
+@media screen and (max-width: 39.9375em) {
+  section.additional-info table.table tbody tr td.dataset-details {
+    width: 40%;
+  }
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1262]:
```suggestion
  }
  .label-div {
    width: 15%;
  }
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 12]
```diff
@@ -1081,6 +1193,178 @@ via submission form. Copied from DS.*/
...
+  padding: 1em 0.2em;
+  font-size: 90%;
+}
+
+.resource-item-description{
+  font-weight: 400;
+  font-size: 18px;
+  line-height: 24.51px;
+  color: #1a1a1a;
+}
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1225]:
```suggestion
}
.resource-item.row, .resource-list {
  margin: 0 -15px;
}
.resource-item div {
  padding: unset;
}
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 13]
```diff
@@ -1081,6 +1193,178 @@ via submission form. Copied from DS.*/
...
+  color: #4d4d4d;
+}
+section.additional-info table.table tbody tr th,
+section.additional-info table.table tbody tr td {
+  border: none;
+}
+
+@media screen and (min-width: 73em) {
+
+}
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1342]: remove this as it is empty

**Review by Alexandra-Green — CHANGES_REQUESTED** [review]
This is for the label height and width, after applying these changes the description will be under the label, so you'll have to add a solution for that

File: ckanext/ontario_theme/fanstatic/common.css [hunk 14]
```diff
@@ -1081,6 +1193,178 @@ via submission form. Copied from DS.*/
...
+.label.label-french,
+.label.label-english_and_french {
+  background-color: #f2f2f2;
+  font-size: 20px;
+  font-weight: 700;
+  color: #1a1a1a;
+  border-radius: 2px;
+  margin-left: 10px;
+  margin-bottom: 5px;
+}
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1236]:
```suggestion
.label.label-english,
.label.label-french,
.label.label-english_and_french {
  background-color: #f2f2f2;
  font-size: 20px;
  font-weight: 700;
  color: #1a1a1a;
  border-radius: 2px;
  margin-left: 10px;
  margin-bottom: 5px;
  vertical-align: -30%;
}
.description.details {
  margin-top: 10px;
}
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 15]
```diff
@@ -1081,6 +1193,178 @@ via submission form. Copied from DS.*/
...
+
+.resource-item-title{
+  font-weight: 700;
+  font-size: 20px;
+  line-height: 27px;
+  color: #000000;
+  text-decoration: none;
+}
+
```
Comment (Alexandra-Green) [inline ckanext/ontario_theme/fanstatic/common.css:1245]:
```suggestion
.resource-item-title{
  font-weight: 700;
  font-size: 20px;
  line-height: 27px;
  color: #000000;
  text-decoration: none;
  vertical-align: -30%;
}

```

**Review by Alexandra-Green — CHANGES_REQUESTED** [review]
This is for the wrapping of the english and french labels

File: ckanext/ontario_theme/fanstatic/common.css (hunk 14 above)
Comment (rleblanc-on) [reply ckanext/ontario_theme/fanstatic/common.css:1236]: Works great!

File: ckanext/ontario_theme/fanstatic/common.css (hunk 15 above)
Comment (rleblanc-on) [reply ckanext/ontario_theme/fanstatic/common.css:1245]: Looks good!

**Comment by KatiRG** [comment]
The word wrapping is better but still breaking long words awkwardly:
Screen Shot 2023-03-19 at 16 02 15

**Review by KatiRG — APPROVED** [review]
The designers can add any UI tweaks on their next round.

**Pull request merged**